python/ray/dag/compiled_dag_node.py

```diff
@@ -746,6 +743,16 @@ def _get_node_id(self):
             for idx in task.downstream_node_idxs:
                 frontier.append(idx)
 
+        from ray.dag.constants import RAY_ADAG_ENABLE_VERIFY_GRAPH
+
+        if RAY_ADAG_ENABLE_VERIFY_GRAPH and not self._detect_deadlock():
+            raise ValueError(
+                "This DAG cannot be compiled because it will deadlock on NCCL "
+                "calls. If this is a false positive, please disable the graph "
```

Comment:
Ideally, it shouldn't have false positives. However, I still provide a feature flag for users to turn it off because if there are false positives, users would need to wait for us to fix and release the next version. The feature flag offers users a workaround.

Reply:
Maybe update the wording just to say something like "If you believe this is a false positive"

Reply:
Update https://github.com/ray-project/ray/pull/45960/commits/babd559e0a1e300aa7f96210482d55a97ff0eb1c